From 170834fe1d0fca7226f3a2f343482d5bf4459b30 Mon Sep 17 00:00:00 2001
From: Pavan Nikhilesh <pbhagavatula@marvell.com>
Date: Fri, 23 Apr 2021 02:03:33 +0530
Subject: [PATCH 1490/1921] octeontx2-af: ratelimit digest prints

Ratelimit SSO AF error interrupt digest prints.

Signed-off-by: Pavan Nikhilesh <pbhagavatula@marvell.com>
Change-Id: I6b7f930608efca24848703c5fea86dbe34a391ae
Reviewed-on: https://sj1git1.cavium.com/c/IP/SW/kernel/linux/+/50535
Tested-by: sa_ip-sw-jenkins <sa_ip-sw-jenkins@marvell.com>
Reviewed-by: Sunil Kovvuri Goutham <sgoutham@marvell.com>
[WK: The original patch got from Marvell sdk11.21.09]
Signed-off-by: Wenlin Kang <wenlin.kang@windriver.com>
---
 drivers/net/ethernet/marvell/octeontx2/af/rvu_sso.c | 3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

diff --git a/drivers/net/ethernet/marvell/octeontx2/af/rvu_sso.c b/drivers/net/ethernet/marvell/octeontx2/af/rvu_sso.c
index 6fe935c94b00..9fdf53eac3af 100644
--- a/drivers/net/ethernet/marvell/octeontx2/af/rvu_sso.c
+++ b/drivers/net/ethernet/marvell/octeontx2/af/rvu_sso.c
@@ -36,7 +36,8 @@
 #define SSO_AF_INT_DIGEST_PRNT(reg)					\
 	for (i = 0; i < block->lf.max / 64; i++) {			\
 		reg0 = rvu_read64(rvu, blkaddr, reg##X(i));		\
-		dev_err(rvu->dev, #reg "(%d) : 0x%llx", i, reg0);	\
+		dev_err_ratelimited(rvu->dev, #reg "(%d) : 0x%llx", i,  \
+				    reg0);				\
 		rvu_write64(rvu, blkaddr, reg##X(i), reg0);		\
 	}
 
-- 
2.31.1

